CommonMC/src/CosmicTimeOffset_module.cc

```diff
@@ -55,13 +60,16 @@ namespace mu2e {
   //================================================================
   CosmicTimeOffset::CosmicTimeOffset(const Parameters& conf)
     : EDProducer{conf}
-    , engine_(createEngine(art::ServiceHandle<SeedService>()->getSeed()) )
+    , engine_(createEngine(art::ServiceHandle<SeedService>()->getSeed()))
+    , cosmicModuleLabel_(conf().cosmicModuleLabel())
+    , addTimeOffset_(conf().addTimeOffset())
     , verbosityLevel_(conf().verbosityLevel())
     , intervalStart_(conf().intervalStart())
     , intervalEnd_(conf().intervalEnd())
     , flatTime_(engine_, intervalStart_, intervalEnd_)
   {
     produces<SimTimeOffset>();
+    produces<GenParticleCollection>();
```

Comment:
This produces should be conditional on addTimeOffset_.  Also you need a mayConsume on the input GenParticle collection.

Reply:
Ok